app/projects/steps/dynamic_form.py

```diff
@@ -0,0 +1,112 @@
+from abc import ABC, abstractmethod
+from projects.models import SignedAgreementForm
+from projects.models import AGREEMENT_FORM_TYPE_STATIC
+
+from projects.forms.payerdb import AccessRequestForm
+
+from projects.models import AgreementForm
+from projects.models import DataProject
+from projects.models import PayerDBForm
+
+from datetime import datetime
+
+class ProjectStep:
+
+    template = None
+    status = None
+    agreement_form = None
+    return_url = None
+    model_name = None
+
+    def __init__(self, title, project):
+        self.title = title
+        self.project = project
+
+    def __str__(self):
+        return "title : %s project : %s" % (self.title, self.project)
+
+
+def save_dynamic_form(agreement_form_id, project_key, model_name, posted_form, user):
+    agreement_form = AgreementForm.objects.get(id=agreement_form_id)
+    project = DataProject.objects.get(project_key=project_key)
+
+    payer_db_form_type = agreement_form_factory(model_name)
```

Comment:
Other types of projects besides the Payer DB project would use this function right? Would be good to change the name if so to prevent confusion.

Reply:
Good catch, will update in next push.